plugins/wme-sitebuilder/wme-sitebuilder/Plugins/PaymentGateways/PayPal.php

```diff
@@ -71,20 +71,27 @@ public function registerHooks() {
 	 * @return array
 	 */
 	public function card_row_props() {
+		$button_props = [
+			'label'           => __( 'Connect PayPal', 'wme-sitebuilder' ),
+			'backgroundColor' => '#172C70',
+			'href'            => add_query_arg( 'page', 'sitebuilder-store-details#/wizard/payments-paypal', admin_url( 'admin.php' ) ),
+		];
+
+		if ( $this->connected ) {
+			$button_props = [
+				'label'           => __( 'Manage PayPal', 'wme-sitebuilder' ),
+				'backgroundColor' => '#000000',
+				'href'            => $this->admin_url,
+			];
+		}
+
```

Comment:
@bswatson There's probably a better way of handling this?

Reply:
What about setting up all of the possible options and letting the state of `$this-connected` determine which one is passed through? Would look something like:

```php
		$button_props = [
			'setup' => [
				'label'           => __( 'Connect PayPal', 'wme-sitebuilder' ),
				'backgroundColor' => '#172C70',
				'href'            => add_query_arg( 'page', 'sitebuilder-store-details#/wizard/payments-paypal', admin_url( 'admin.php' ) ),
			],
			'connected' => [
				'label'           => __( 'Manage PayPal', 'wme-sitebuilder' ),
				'backgroundColor' => '#000000',
				'href'            => $this->admin_url,
			]
		];

		return [
			'id'        => 'payments-paypal',
			'type'      => 'task',
			'title'     => __( 'Set Up PayPal', 'wme-sitebuilder' ),
			'intro'     => __( 'Receive payments via PayPal.', 'wme-sitebuilder' ),
			'connected' => $this->connected,
			'button'    => $button_props[ $this->connected ? 'connected' : 'setup' ],
		];
```

Another option would be to send all of the possible states through with the props and let the Front End decide which one to display based on the value of `props.connected`. That would then allow the button to update without requiring a page refresh.

Reply:
I liked the code example you shared above. It felt more in line with what we currently do.